### PR TITLE
Jlinoff 20210107 docstyle

### PR DESCRIPTION
**Pull request by jlinoff**
1. Convert source code documents to google style.
2. Added port reporting to status.
3. Add mypy type checking.

### CODE REVIEW

**Comment by jlinoff** [comment]
@Irialad , since this change was mostly just documentation and type check changes, you probably olyn need to look at status.py.

File: grape/common/log.py [hunk 1]
```diff
@@ -130,35 +134,39 @@ def warn(msg: str, level: int = 1):
...
+    assert LOG  # make sure LOG is initialized
     try:
         LOG.debug(msg, stacklevel=level+1, extra=extra())
     except ValueError:  # recover from pylint IO issue
         init(LEVEL)
         LOG.debug(msg, stacklevel=level+1, extra=extra())
 
 
-def err(msg: str, level: int = 1, xflag=True):
+def err(msg: str, level: int = 1, xflag=True, xcode=1):
```
[the remark this answers is not in the document]
Comment (jlinoff) [reply grape/common/log.py]: It is utterly odd and an oversight! Thanks for catching that.

[the remark this answers is not in the document]
Comment (jlinoff) [reply grape/common/log.py]: Fixed.

File: grape/status.py [hunk 2]
```diff
@@ -102,8 +103,13 @@ def hdr(self, prefix: str = '  '):
...
 def get_elapsed_time(start_str: str) -> str:
-    '''
-    Get the elapsed time in seconds.
+    '''Get the elapsed time in seconds.
+
+    Args:
+        start_str: The container start timestamp.
+
+    Returns:
+        exlapsed: The elapsed time as a string.
```
[the remark this answers is not in the document]
Comment (jlinoff) [reply grape/status.py]: I wonder what i was thinking.
Fixed.

**Review by Irialad — APPROVED** [review]

**Pull request merged**